framework/source/class/qx/bom/request/Xhr.js

```diff
@@ -922,15 +934,19 @@ qx.Bootstrap.define("qx.bom.request.Xhr",
       this.status = 0;
       this.statusText = this.responseText = "";
       this.responseXML = null;
+      this.response = null;
 
       if (this.readyState >= qx.bom.request.Xhr.HEADERS_RECEIVED) {
         // In some browsers, XHR properties are not readable
         // while request is in progress.
         try {
           this.status = nxhr.status;
           this.statusText = nxhr.statusText;
-          this.responseText = nxhr.responseText;
-          this.responseXML = nxhr.responseXML;
+          this.response = nxhr.response;
+          if (typeof nxhr.response === "string") {
```

Comment:
Yes, if responseTyp is not empty or not "text" an object will be returned.
https://xhr.spec.whatwg.org/#the-responsetype-attribute